services/src/datasets/storage.rs

```diff
@@ -0,0 +1,273 @@
+use crate::datasets::listing::DataSetProvider;
+use crate::error::Result;
+use crate::projects::project::LayerInfo;
+use crate::users::user::UserId;
+use crate::util::user_input::{UserInput, Validated};
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+use geoengine_datatypes::collections::{FeatureCollection, TypedFeatureCollection};
+use geoengine_datatypes::dataset::{
+    DataSetId, DataSetProviderId, InternalDataSetId, StagingDataSetId,
+};
+use geoengine_datatypes::identifiers::Identifier;
+use geoengine_datatypes::primitives::Geometry;
+use geoengine_datatypes::raster::{Pixel, RasterTile2D};
+use geoengine_operators::engine::{RasterResultDescriptor, VectorResultDescriptor};
+use geoengine_operators::mock::MockDataSetDataSourceLoadingInfo;
+use geoengine_operators::source::OgrSourceDataset;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+use std::path::PathBuf;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct DataSet {
+    pub id: DataSetId,
+    pub name: String,
+    pub description: String,
+    pub data_type: LayerInfo, // TODO: custom type instead of reusing existing one?
```

Comment:
I don't know, maybe layer = dataset. We just have to be cautious to not have conflicting updates.

Reply:
what is a Layer in this context?

Reply:
here I want to store the data type and/or columns of the data set. The same infos that we need for layer so I was thinking about sharing the struct.